db_stress_tool/expected_value.h

```diff
@@ -160,7 +163,51 @@ class PendingExpectedValue {
         orig_value_(orig_value),
         final_value_(final_value) {}
 
+  PendingExpectedValue(const PendingExpectedValue& other)
+      : value_ptr_(other.value_ptr_),
+        orig_value_(other.orig_value_),
+        final_value_(other.final_value_) {
+    other.ClosePendingState();
+  }
+
+  PendingExpectedValue(PendingExpectedValue&& other)
+      : value_ptr_(std::move(other.value_ptr_)),
+        orig_value_(std::move(other.orig_value_)),
+        final_value_(std::move(other.final_value_)) {
+    other.ClosePendingState();
+  }
+
+  PendingExpectedValue& operator=(const PendingExpectedValue& other) {
+    if (this != &other) {
+      other.ClosePendingState();
+      value_ptr_ = other.value_ptr_;
+      orig_value_ = other.orig_value_;
+      final_value_ = other.final_value_;
+    }
+    return *this;
+  }
+
+  PendingExpectedValue& operator=(PendingExpectedValue&& other) {
+    if (this != &other) {
+      other.ClosePendingState();
+      value_ptr_ = std::move(other.value_ptr_);
+      orig_value_ = std::move(other.orig_value_);
+      final_value_ = std::move(other.final_value_);
+    }
+    return *this;
+  }
```

Comment:
Why do we need a full-fledged class?

Reply:
This is similar to the `Status` class when we want to make sure it's checked, but once `Status` is created it can be assigned to another `Status` or copied over to another `Status`. The original copied-from or moved-from `Status` need to have its checked flag explicitly flipped in a constructor or assignment operator, so that its destructor wouldn't assert.

One util method creating a `PendingExpectedValue` and then pass it along to the actual test site, or one `PendingExpectedValue` getting assigned to another `PendingExpectedValue` are common patterns we have.